### ISSUE TITLE
Question about "looseIndexOf" in ui-select

### ISSUE THREAD

**Issue by macgyver** [JosephusPaye/Keen-UI#451]
Since there is already a `key` specified for the options, would it not make more sense to just compare the keys of the objects in the [`isOptionSelected`](https://github.com/JosephusPaye/Keen-UI/blob/master/src/UiSelect.vue#L463) method?

**Comment by JosephusPaye** [comment]
Hi @macgyver,

That's a good point, the component could use `key` from options if it's available (not always, as `key` is optional).

For cases where it's not available, we need to keep `looseIndexOf`, or something similar, since comparing objects directly by reference could have false negatives.

**Comment by macgyver** [comment]
It's optional as a prop, but it seems like the keys are always defined, with sensible defaults:
https://github.com/JosephusPaye/Keen-UI/blob/master/src/UiSelect.vue#L232

**Comment by JosephusPaye** [comment]
That's true, but the problem with depending on that is that when the user sets the `keys` prop, it overrides the defaults, so if they don't specify all the keys, e.g. 

```js
{ class: 'my-class', image: 'my_image_field' }
```

you end up with no `id` key.

This might be less of an issue in practice though - so we can check if `keys.id` is available and use that to compare, falling back to `looseIndexOf` if not.

**Comment by macgyver** [comment]
Ah, I didn't understand how the defaults was being used - I assumed incorrectly that any specified keys would be assigned over the defaults. At any rate I think you're right, checking for the presence of the "id" key specifically before using it for comparison should yield the desired effect.

If you're curious about my use-case, I'm storing the selected results of the ui-select as an object but weeks later when I go to re-populate the select the label of any of the options may have been edited (only the value field is immutable). I could update my selected value object manually before sending it to the ui-select so that it matches up perfectly with one of the options, which is how I'm getting around this right now, but I think it will be slightly faster to do it in the UI framework, and would also help if somehow the key order got mixed up but the shape of the object was still the same.

**Comment by JosephusPaye** [comment]
Makes sense, will add.
